File: jcv-core/src/main/java/com/ekino/oss/jcv/core/comparator/UUIDComparator.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator;

import java.util.UUID;

import com.ekino.oss.jcv.core.JsonValueComparator;

import org.skyscreamer.jsonassert.ValueMatcherException;

public class UUIDComparator implements JsonValueComparator<String> {

    @Override
    public boolean hasCorrectValue(String actual, String expected) {
        try {
            //noinspection ResultOfMethodCallIgnored
            UUID.fromString(actual);
            return true;
        } catch (IllegalArgumentException e) {
            throw new ValueMatcherException("Value is not a valid UUID", e, expected, actual);
        }
    }
}

File: jcv-core/src/main/kotlin/com/ekino/oss/jcv/core/comparator/UUIDComparator.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator

import com.ekino.oss.jcv.core.JsonValueComparator
import org.skyscreamer.jsonassert.ValueMatcherException
import java.util.UUID

class UUIDComparator : JsonValueComparator<String> {

    override fun hasCorrectValue(actual: String?, expected: String?): Boolean {
        try {
            UUID.fromString(actual)
            return true
        } catch (e: IllegalArgumentException) {
            throw ValueMatcherException("Value is not a valid UUID", e, expected, actual)
        }
    }
}
